File: voyage/ColorPalette.swift
import SwiftUI
import UIKit

/// Centralized color palette for the app
/// All colors are defined once here and referenced throughout the codebase
enum AppColors {
    // MARK: - Map/Globe Colors

    /// Ocean blue - #2F86A6
    static let ocean = Color(red: 0.184, green: 0.525, blue: 0.651)
    static let oceanUI = UIColor(red: 0.184, green: 0.525, blue: 0.651, alpha: 1.0)

    /// Land green - #34BE82
    static let land = Color(red: 0.204, green: 0.745, blue: 0.510)
    static let landUI = UIColor(red: 0.204, green: 0.745, blue: 0.510, alpha: 1.0)

    /// Land when selected (brighter green)
    static let landSelected = Color(red: 0.45, green: 0.85, blue: 0.60)
    static let landSelectedUI = UIColor(red: 0.45, green: 0.85, blue: 0.60, alpha: 1.0)

    /// Visited yellow - #F2F013
    static let visited = Color(red: 0.949, green: 0.941, blue: 0.075)
    static let visitedUI = UIColor(red: 0.949, green: 0.941, blue: 0.075, alpha: 1.0)

    /// Visited when selected (brighter yellow)
    static let visitedSelected = Color(red: 1.0, green: 1.0, blue: 0.3)
    static let visitedSelectedUI = UIColor(red: 1.0, green: 1.0, blue: 0.3, alpha: 1.0)

    /// Wishlist purple
    static let wishlist = Color(red: 0.6, green: 0.4, blue: 0.8)
    static let wishlistUI = UIColor(red: 0.6, green: 0.4, blue: 0.8, alpha: 1.0)

    /// Wishlist when selected (brighter purple)
    static let wishlistSelected = Color(red: 0.75, green: 0.55, blue: 0.95)
    static let wishlistSelectedUI = UIColor(red: 0.75, green: 0.55, blue: 0.95, alpha: 1.0)

    /// Atmosphere glow
    static let atmosphere = UIColor(red: 0.6, green: 0.8, blue: 1.0, alpha: 0.15)

    /// Dark mode ocean
    static let oceanDark = Color(red: 0.1, green: 0.15, blue: 0.25)

    // MARK: - UI Button Colors

    /// Primary button color (light mode) - warm orange
    static let buttonLight = Color(red: 0.85, green: 0.55, blue: 0.35)

    /// Primary button color (dark mode) - muted purple
    static let buttonDark = Color(red: 0.4, green: 0.35, blue: 0.6)

    /// Visited/success button green
    static let buttonVisited = Color(red: 0.3, green: 0.7, blue: 0.4)

    // MARK: - UI Background Colors

    /// Light mode warm gradient top
    static let backgroundLightTop = Color(red: 0.98, green: 0.96, blue: 0.93)

    /// Light mode warm gradient bottom
    static let backgroundLightBottom = Color(red: 0.95, green: 0.91, blue: 0.87)

    /// Dark mode card/panel background
    static let cardDark = Color(red: 0.2, green: 0.2, blue: 0.25)

    /// Dark mode secondary background
    static let cardDarkSecondary = Color(red: 0.15, green: 0.15, blue: 0.2)

    /// Light mode track/divider
    static let trackLight = Color(red: 0.9, green: 0.88, blue: 0.85)

    /// Dark mode track/divider
    static let trackDark = Color(red: 0.25, green: 0.25, blue: 0.3)

    /// Close button background (dark mode)
    static let closeButtonDark = Color(red: 0.3, green: 0.3, blue: 0.35)

    /// Close button background (light mode)
    static let closeButtonLight = Color(red: 0.9, green: 0.9, blue: 0.9)

    // MARK: - UI Text Colors

    /// Primary text (light mode) - warm brown
    static let textPrimaryLight = Color(red: 0.2, green: 0.15, blue: 0.1)

    /// Secondary text (light mode)
    static let textSecondaryLight = Color(red: 0.4, green: 0.35, blue: 0.3)

    /// Tertiary text (light mode)
    static let textTertiaryLight = Color(red: 0.5, green: 0.45, blue: 0.4)

    /// Muted text (light mode)
    static let textMutedLight = Color(red: 0.6, green: 0.55, blue: 0.5)

    /// Secondary text (dark mode)
    static let textSecondaryDark = Color(red: 0.7, green: 0.7, blue: 0.75)

    /// Tertiary text (dark mode)
    static let textTertiaryDark = Color(red: 0.6, green: 0.6, blue: 0.65)

    /// Muted text (dark mode)
    static let textMutedDark = Color(red: 0.5, green: 0.5, blue: 0.55)

    /// Close button text (light mode)
    static let closeButtonText = Color(red: 0.3, green: 0.3, blue: 0.3)

    // MARK: - Progress Bar Colors

    /// Progress gradient (dark mode)
    static let progressDarkStart = Color(red: 0.5, green: 0.4, blue: 0.8)
    static let progressDarkEnd = Color(red: 0.6, green: 0.5, blue: 0.9)

    /// Progress gradient (light mode)
    static let progressLightStart = Color(red: 0.85, green: 0.5, blue: 0.3)
    static let progressLightEnd = Color(red: 0.95, green: 0.6, blue: 0.4)

    // MARK: - Page Background Colors

    /// Page background (dark mode)
    static let pageBgDark = Color(red: 0.1, green: 0.1, blue: 0.12)

    /// Page background (light mode)
    static let pageBgLight = Color(red: 0.96, green: 0.95, blue: 0.93)

    /// Badge text (dark mode)
    static let badgeTextDark = Color(red: 0.8, green: 0.8, blue: 0.85)

    /// Badge text (light mode)
    static let badgeTextLight = Color(red: 0.3, green: 0.25, blue: 0.2)

    // MARK: - Helper Functions

    /// Returns the appropriate button color based on dark mode
    static func buttonColor(isDarkMode: Bool) -> Color {
        isDarkMode ? buttonDark : buttonLight
    }

    /// Returns the appropriate text color based on dark mode
    static func textPrimary(isDarkMode: Bool) -> Color {
        isDarkMode ? .white : textPrimaryLight
    }

    /// Returns the appropriate secondary text color based on dark mode
    static func textSecondary(isDarkMode: Bool) -> Color {
        isDarkMode ? textSecondaryDark : textSecondaryLight
    }

    /// Returns the appropriate tertiary text color based on dark mode
    static func textTertiary(isDarkMode: Bool) -> Color {
        isDarkMode ? textTertiaryDark : textTertiaryLight
    }

    /// Returns the appropriate muted text color based on dark mode
    static func textMuted(isDarkMode: Bool) -> Color {
        isDarkMode ? textMutedDark : textMutedLight
    }

    /// Returns the appropriate track color based on dark mode
    static func track(isDarkMode: Bool) -> Color {
        isDarkMode ? trackDark : trackLight
    }

    /// Returns the appropriate card background based on dark mode
    static func cardBackground(isDarkMode: Bool) -> Color {
        isDarkMode ? cardDark : .white
    }

    /// Returns the appropriate page background based on dark mode
    static func pageBackground(isDarkMode: Bool) -> Color {
        isDarkMode ? pageBgDark : pageBgLight
    }
}

File: android/app/src/main/kotlin/com/anmol/voyage/ui/theme/ColorPalette.kt
package com.anmol.voyage.ui.theme

import androidx.compose.ui.graphics.Color

/**
 * Centralized color palette for the app — the Android counterpart of the iOS
 * `AppColors` enum (`ios/voyage/ColorPalette.swift`).
 *
 * Every value here is the exact same color as its iOS twin (the iOS file stores
 * fractional sRGB components; the hex literals below are those components
 * scaled to 8 bits). Colors are defined once here and referenced throughout the
 * codebase — never hardcode a hex value elsewhere, and when a color changes on
 * one platform, change it on the other in the same PR.
 *
 * Country-status colors (land / visited / wishlist / selected variants) are
 * *semantic*: they must never be replaced by Material You dynamic colors, which
 * apply to app chrome only. See [VoyageTheme].
 *
 * Note on ordering: colors that derive from another color are declared after it,
 * because properties of a Kotlin object initialize in declaration order.
 */
object VoyagePalette {

    // region UI button colors

    /** Primary button color — warm orange #D98C59 */
    val buttonColor = Color(0xFFD98C59)

    /** Visited/success button green */
    val buttonVisited = Color(0xFF4CB266)

    // endregion

    // region Map/globe colors

    /** Ocean blue — #2F86A6 */
    val ocean = Color(0xFF2F86A6)

    /** Land green (unvisited) — #34BE82 */
    val land = Color(0xFF34BE82)

    /** Land when selected (brighter green) */
    val landSelected = Color(0xFF73D999)

    /** Visited yellow — #F2F013 */
    val visited = Color(0xFFF2F013)

    /** Visited when selected (brighter yellow) */
    val visitedSelected = Color(0xFFFFFF4C)

    /** Wishlist purple */
    val wishlist = Color(0xFF9966CC)

    /** Wishlist when selected (brighter purple) */
    val wishlistSelected = Color(0xFFBF8CF2)

    /** Atmosphere glow (15% alpha, as on iOS) */
    val atmosphere = Color(0x2699CCFF)

    /** Ocean for the flat map view (slightly darker than the globe's) */
    val oceanMap = Color(0xFF31739B)

    /** Dark mode ocean */
    val oceanDark = Color(0xFF1A2640)

    /** Capital star marker — reuses the theme orange, as on iOS */
    val capitalMarker = buttonColor

    /** Thin rim stroked around the capital star to lift it off the country fill */
    val capitalMarkerOutline = Color(0xFF4C4C4C)

    // endregion

    // region Background colors

    /** Light mode warm gradient top */
    val backgroundLightTop = Color(0xFFFAF5ED)

    /** Light mode warm gradient bottom */
    val backgroundLightBottom = Color(0xFFF2E8DE)

    /** Dark mode card/panel background */
    val cardDark = Color(0xFF333340)

    /** Dark mode secondary background */
    val cardDarkSecondary = Color(0xFF262633)

    /** Light mode track/divider */
    val trackLight = Color(0xFFE6E0D9)

    /** Dark mode track/divider */
    val trackDark = Color(0xFF40404C)

    /** Close button background (dark mode) */
    val closeButtonDark = Color(0xFF4C4C59)

    /** Close button background (light mode) */
    val closeButtonLight = Color(0xFFE6E6E6)

    /** Page background (dark mode) */
    val pageBgDark = Color(0xFF1A1A1F)

    /** Page background (light mode) */
    val pageBgLight = Color(0xFFF5F2ED)

    // endregion

    // region Text colors

    /** Primary text (light mode) — warm brown */
    val textPrimaryLight = Color(0xFF33261A)

    /** Secondary text (light mode) */
    val textSecondaryLight = Color(0xFF66594C)

    /** Tertiary text (light mode) */
    val textTertiaryLight = Color(0xFF807366)

    /** Muted text (light mode) */
    val textMutedLight = Color(0xFF998C80)

    /** Secondary text (dark mode) */
    val textSecondaryDark = Color(0xFFB2B2BF)

    /** Tertiary text (dark mode) */
    val textTertiaryDark = Color(0xFF9999A6)

    /** Muted text (dark mode) */
    val textMutedDark = Color(0xFF80808C)

    /** Close button text (light mode) */
    val closeButtonText = Color(0xFF4C4C4C)

    /** Badge text (dark mode) */
    val badgeTextDark = Color(0xFFCCCCD9)

    /** Badge text (light mode) */
    val badgeTextLight = Color(0xFF4C4033)

    // endregion

    // region Progress bar colors

    /** Progress gradient (dark mode) */
    val progressDarkStart = Color(0xFF8066CC)
    val progressDarkEnd = Color(0xFF9980E6)

    /** Progress gradient (light mode) */
    val progressLightStart = Color(0xFFD9804C)
    val progressLightEnd = Color(0xFFF29966)

    // endregion

    // region Challenge game colors

    /** Correct guess fill (deep green, distinct from unvisited land) */
    val challengeCorrect = Color(0xFF1A8C40)

    /** Missed country fill and wrong-guess/reveal banners */
    val challengeWrong = Color(0xFFD94033)

    // endregion

    // region Medal colors (achievement medals)

    /** Gold face gradient (unlocked medal) */
    val medalGoldCenter = Color(0xFFFFDE6B)
    val medalGoldEdge = Color(0xFFC78F26)
    val medalGoldRim = Color(0xFFA6731A)

    /** Silver face gradient (locked medal) */
    val medalSilverCenter = Color(0xFFD1D1D6)
    val medalSilverEdge = Color(0xFF8C8C94)
    val medalSilverRim = Color(0xFF73737A)

    // endregion

    // region Trophy colors (challenge trophies)

    /** Bronze trophy gradient */
    val trophyBronzeLight = Color(0xFFD98F4C)
    val trophyBronzeDark = Color(0xFF8C5426)

    /** Silver/gold trophy gradients reuse the medal palette */
    val trophySilverLight = medalSilverCenter
    val trophySilverDark = medalSilverEdge
    val trophyGoldLight = medalGoldCenter
    val trophyGoldDark = medalGoldEdge

    // endregion

    // region Dark-mode helpers (mirror the iOS helper functions)

    fun textPrimary(isDark: Boolean): Color = if (isDark) Color.White else textPrimaryLight

    fun textSecondary(isDark: Boolean): Color = if (isDark) textSecondaryDark else textSecondaryLight

    fun textTertiary(isDark: Boolean): Color = if (isDark) textTertiaryDark else textTertiaryLight

    fun textMuted(isDark: Boolean): Color = if (isDark) textMutedDark else textMutedLight

    fun track(isDark: Boolean): Color = if (isDark) trackDark else trackLight

    fun cardBackground(isDark: Boolean): Color = if (isDark) cardDark else Color.White

    fun pageBackground(isDark: Boolean): Color = if (isDark) pageBgDark else pageBgLight

    // endregion
}
